### PR TITLE
feat(showcase): add typescript tooling

### PR DESCRIPTION
**Pull request by aklkv**
### :pushpin: Summary

* add typescript tooling
* update app blueprint
* fix in what order GHA runs build/lint/test for showcase as we need components and flight-icons to be built

### :link: External links

Jira ticket: [HDS-XXX](https://hashicorp.atlassian.net/browse/HDS-XXX)
Figma file: [if it applies]

***

### 👀 Component checklist

- [ ] Percy was checked for any visual regression
- [ ] A changelog entry was added via [Changesets](https://github.com/changesets/changesets) if needed (see [templates here](https://github.com/hashicorp/design-system/blob/main/wiki/Website-Changelog.md#templates-for-npm-packages))

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

### CODE REVIEW

File: showcase/.prettierrc.js
Comment (didoo) [inline showcase/.prettierrc.js:1]: [blocker?] are these changes intentional? 🤔

File: showcase/blueprints/hds-component-test/index.js [hunk 1]
```diff
@@ -4,7 +4,7 @@
  */
 
 /* eslint-disable ember/no-string-prototype-extensions */
-/* eslint-disable node/no-extraneous-require */
+/* eslint-disable n/no-extraneous-require */
```
Comment (didoo) [inline showcase/blueprints/hds-component-test/index.js:7]: 😮

File: showcase/.ember-cli [hunk 2]
```diff
@@ -8,8 +8,8 @@
   "disableAnalytics": true,
 
   /**
-  Setting `isTypeScriptProject` to true will force the blueprint generators to generate TypeScript
-  rather than JavaScript by default, when a TypeScript version of a given blueprint is available.
+    Setting `isTypeScriptProject` to true will force the blueprint generators to generate TypeScript
+    rather than JavaScript by default, when a TypeScript version of a given blueprint is available.
   */
-  "isTypeScriptProject": false
+  "isTypeScriptProject": true
```
Comment (didoo) [inline showcase/.ember-cli:14]: @alex-ju since we are enabling the blueprints for the showcase as TS, do you think we should also update the `showcase/blueprints/hds-component/files/src/components/hds/__name__/index.js` file to remove the `Index` from the class name (in this PR, I mean)?

**Review by didoo — COMMENTED** [review]
@aklkv I've left a few comments (one of them, related to the `showcase/.prettierrc.js` file, makes me think it may be accidental?) but please wait for more TS-knowledgeable people like @alex-ju @natmegs to approve before merging ;)

File: showcase/.ember-cli (hunk 2 above)
Comment (alex-ju) [reply showcase/.ember-cli:14]: yes, [we have an action for this already](https://docs.google.com/document/d/1twnk39hZ7p8oZYrld_FYsdiuUb6yn_weA46KnhF4nu4/edit?disco=AAABNdTuLbc)

File: .yarn/patches/@glint-environment-ember-loose-npm-1.4.0-31c2f31bcb.patch [hunk 3]
```diff
@@ -0,0 +1,15 @@
+diff --git a/-private/environment/index.js b/-private/environment/index.js
+index 217583f4bae53f6f72db8d3debd0f31b3dac2b8e..aa2c1a2a665005d8c95b9e3018640fdf7e317861 100644
+--- a/-private/environment/index.js
++++ b/-private/environment/index.js
+@@ -46,10 +46,6 @@ function emberLooseEnvironment(options) {
+                     // Pod component/controller/route
+                     candidates.push(templatePath.replace(REGEXES.POD_TEMPLATE, '/component.ts'), templatePath.replace(REGEXES.POD_TEMPLATE, '/controller.ts'), templatePath.replace(REGEXES.POD_TEMPLATE, '/route.ts'));
+                 }
+-                else if (templatePath.includes('/templates/components/')) {
```
[the remark this answers is not in the document]
Comment (aklkv) [reply .yarn/patches/@glint-environment-ember-loose-npm-1.4.0-31c2f31bcb.patch:9]: This is what has been submitted to the upstream repo as a fix and it's going to go out in the next major https://github.com/typed-ember/glint/pull/721 . I keeping an eye on the new versions once one is released I will follow up with fix PR here

File: showcase/.prettierrc.js
Comment (aklkv) [reply showcase/.prettierrc.js:1]: yes, all this has not been updated in a while and mainly removes deprecated paths that will never come back

File: showcase/blueprints/hds-component-test/index.js (hunk 1 above)
Comment (aklkv) [reply showcase/blueprints/hds-component-test/index.js:7]: `eslint-plugin-node` has been unmaintained for a while and this is the best way the community could come up with, so it's being replaced with `eslint-plugin-n`

File: showcase/.editorconfig
[the remark this answers is not in the document]
Comment (aklkv) [reply showcase/.editorconfig:1]: I have not seen any changes in formatting on my end so I do believe this is a safe change and it will not introduce any charn

File: .yarn/patches/@glint-environment-ember-loose-npm-1.4.0-31c2f31bcb.patch (hunk 3 above)
[the remark this answers is not in the document]
Comment (aklkv) [reply .yarn/patches/@glint-environment-ember-loose-npm-1.4.0-31c2f31bcb.patch:9]: I have update the patch with comments and upstream PR reference

File: showcase/.prettierrc.js
Comment (aklkv) [reply showcase/.prettierrc.js:1]: I have checked back, and what this change does is switch to the new default where `singleQuote` is set to `false` by default we are therefore we are setting `js,ts` explicitly to have a single quote dropping overwrite for `*.scss` files as it's now the default behaviour as well as keeping `printWidth` for all `*.hbs` files, btw does it makes sense 🤔 I feel like default value makes a bit more readable but we can address it separately as well as new ember-template-lint as it has some notable changes related to formatting. Otherwise, this change mainly simplifies and has no effect on how things are formatted

**Review by alex-ju — APPROVED** [review]

**Comment by alex-ju** [comment]
Rebased and regenerated `yarn.lock` to resolve conflicts

File: .yarn/patches/@glint-environment-ember-loose-npm-1.4.0-31c2f31bcb.patch (hunk 3 above)
[the remark this answers is not in the document]
Comment (didoo) [reply .yarn/patches/@glint-environment-ember-loose-npm-1.4.0-31c2f31bcb.patch:9]: Thanks @aklkv 🙏 (also for the upstream PR).

File: showcase/.prettierrc.js
Comment (didoo) [reply showcase/.prettierrc.js:1]: Double checked with @alex-ju, all good.
